src/components/tooltip/Tooltip.tsx

```diff
@@ -217,8 +219,27 @@ class Tooltip extends React.Component<TooltipProps, State> {
             stopBubble,
             text,
             theme,
+            intl,
         } = this.props;
 
+        let ariaLabel;
+        let tooltipText;
+
+        if (text) {
+            if (typeof text === 'string') {
+                tooltipText = text;
+            } else if (React.isValidElement(text) && getProp(text, 'props')) {
+                const { id, defaultMessage, description, values } = getProp(text, 'props');
+                if (id && defaultMessage) {
+                    tooltipText = intl.formatMessage({ id, defaultMessage, description }, values);
+                }
+            }
+        }
+        if (getProp(children, 'props')) {
+            ariaLabel = getProp(children, 'props.aria-label');
```

Comment:
will aria-label always be a prop on children? should we have a default in case it's not or does getProp return a default?

Reply:
yes exactly